### ISSUE TITLE
ice2slice: bug with return value and more 

### ISSUE THREAD

**Issue by bernardnormier** [zeroc-ice/ice#1809]
There are several issues with ice2slice. A good test is to convert https://github.com/mumble-voip/mumble/blob/master/src/murmur/MumbleServer.ice

1. A single return type is not converted properly, e.g. 
```slice
idempotent nameToId(name: string) -> returnValue: int32
```
(the returnValue: is incorrect)

2. proxy parameters are not mapped correct, e.g.
```slice
addCallback(cb: MetaCallback) throws (InvalidCallbackException, InvalidSecretException)
```
Should be `MetaCallbackProxy?`.

3. Docs comments on sequence (and presumably dictionary) are lost.

4. It would be nice to generate additional comments in the output file:
 - the file was generated by ice2slice from Xxx.ice
 - some metadata in the source file (like python:seq:tuple) could not be converted
 - `const int PermissionWhisper = 0x100;` and associated doc-comments were not converted

**Comment by pepone** [comment]
> const int PermissionWhisper = 0x100; and associated doc-comments were not converted

What is the preferred behavior here?  Add a comment or emit a warning.

> some metadata in the source file (like python:seq:tuple) could not be converted

I don't see how anybody would care about python metadata at this point if we don't have an IceRPC python mapping, once we do we can update ice2slice to properly convert it.

**Comment by bernardnormier** [comment]
> > const int PermissionWhisper = 0x100; and associated doc-comments were not converted
> 
> What is the preferred behavior here? Add a comment or emit a warning.

We could do both:
 - emit a warning (no const in IceRPC slice)
 - generate a comment
```slice
    // ice2slice could not convert:
    // const int PermissionWhisper = 0x100;
```

I would drop the doc-comment, if any.

> 
> > some metadata in the source file (like python:seq:tuple) could not be converted
> 
> I don't see how anybody would care about python metadata at this point if we don't have an IceRPC python mapping, once we do we can update ice2slice to properly convert it.

Ok.

**Comment by pepone** [comment]
Fixed in #3082 and #3116
